// json-to-csv-vue-3-typescript/dist/json-to-csv-vue-3-typescript.js
import { defineComponent as ir, computed as ar, openBlock as sr, createElementBlock as or, renderSlot as ur, createTextVNode as fr, toDisplayString as cr } from "vue";
var ve = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function jn(c) {
  return c && c.__esModule && Object.prototype.hasOwnProperty.call(c, "default") ? c.default : c;
}
var an = { exports: {} };
an.exports;
(function(c, O) {
  var A = 200, g = "Expected a function", U = "__lodash_hash_undefined__", W = 1, L = 2, ue = 1 / 0, E = 9007199254740991, C = "[object Arguments]", d = "[object Array]", R = "[object Boolean]", P = "[object Date]", j = "[object Error]", Fe = "[object Function]", et = "[object GeneratorFunction]", he = "[object Map]", Ue = "[object Number]", pe = "[object Object]", Ce = "[object Promise]", ut = "[object RegExp]", Pe = "[object Set]", Se = "[object String]", B = "[object Symbol]", a = "[object WeakMap]", i = "[object ArrayBuffer]", f = "[object DataView]", _ = "[object Float32Array]", m = "[object Float64Array]", S = "[object Int8Array]", F = "[object Int16Array]", ae = "[object Int32Array]", Z = "[object Uint8Array]", w = "[object Uint8ClampedArray]", ne = "[object Uint16Array]", x = "[object Uint32Array]", Ae = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, fe = /^\w*$/, y = /^\./, z = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, ce = /[\\^$.*+?()[\]{}|]/g, V = /\\(\\)?/g, se = /^\[object .+?Constructor\]$/, ie = /^(?:0|[1-9]\d*)$/, h = {};
  h[_] = h[m] = h[S] = h[F] = h[ae] = h[Z] = h[w] = h[ne] = h[x] = !0, h[C] = h[d] = h[i] = h[R] = h[f] = h[P] = h[j] = h[Fe] = h[he] = h[Ue] = h[pe] = h[ut] = h[Pe] = h[Se] = h[a] = !1;
  var b = typeof ve == "object" && ve && ve.Object === Object && ve, N = typeof self == "object" && self && self.Object === Object && self, D = b || N || Function("return this")(), re = O && !O.nodeType && O, H = re && !0 && c && !c.nodeType && c, le = H && H.exports === re, X = le && b.process, G = function() {
    try {
      return X && X.binding("util");
    } catch {
    }
  }(), me = G && G.isTypedArray;
  function we(t, r) {
    for (var o = -1, l = t ? t.length : 0; ++o < l; )
      if (r(t[o], o, t))
        return !0;
    return !1;
  }
  function Ie(t) {
    return function(r) {
      return r == null ? void 0 : r[t];
    };
  }
  function Le(t, r) {
    for (var o = -1, l = Array(t); ++o < t; )
      l[o] = r(o);
    return l;
  }
  function Ee(t) {
    return function(r) {
      return t(r);
    };
  }
  function Y(t, r) {
    return t == null ? void 0 : t[r];
  }
  function ee(t) {
    var r = !1;
    if (t != null && typeof t.toString != "function")
      try {
        r = !!(t + "");
      } catch {
      }
    return r;
  }
  function De(t) {
    var r = -1, o = Array(t.size);
    return t.forEach(function(l, I) {
      o[++r] = [I, l];
    }), o;
  }
  function $(t, r) {
    return function(o) {
      return t(r(o));
    };
  }
  function ke(t) {
    var r = -1, o = Array(t.size);
    return t.forEach(function(l) {
      o[++r] = l;
    }), o;
  }
  var Me = Array.prototype, Ge = Function.prototype, Oe = Object.prototype, be = D["__core-js_shared__"], xe = function() {
    var t = /[^.]+$/.exec(be && be.keys && be.keys.IE_PROTO || "");
    return t ? "Symbol(src)_1." + t : "";
  }(), _e = Ge.toString, Te = Oe.hasOwnProperty, q = Oe.toString, $e = RegExp(
    "^" + _e.call(Te).replace(ce, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
  ), ft = D.Symbol, It = D.Uint8Array, qt = Oe.propertyIsEnumerable, on = Me.splice, un = $(Object.keys, Object), Rt = ct(D, "DataView"), tt = ct(D, "Map"), Pt = ct(D, "Promise"), dt = ct(D, "Set"), nt = ct(D, "WeakMap"), rt = ct(Object, "create"), Lt = at(Rt), Dt = at(tt), ht = at(Pt), fn = at(dt), cn = at(nt), wt = ft ? ft.prototype : void 0, Mt = wt ? wt.valueOf : void 0, Wt = wt ? wt.toString : void 0;
  function Ne(t) {
    var r = -1, o = t ? t.length : 0;
    for (this.clear(); ++r < o; ) {
      var l = t[r];
      this.set(l[0], l[1]);
    }
  }
  function Ft() {
    this.__data__ = rt ? rt(null) : {};
  }
  function jt(t) {
    return this.has(t) && delete this.__data__[t];
  }
  function it(t) {
    var r = this.__data__;
    if (rt) {
      var o = r[t];
      return o === U ? void 0 : o;
    }
    return Te.call(r, t) ? r[t] : void 0;
  }
  function ln(t) {
    var r = this.__data__;
    return rt ? r[t] !== void 0 : Te.call(r, t);
  }
  function dn(t, r) {
    var o = this.__data__;
    return o[t] = rt && r === void 0 ? U : r, this;
  }
  Ne.prototype.clear = Ft, Ne.prototype.delete = jt, Ne.prototype.get = it, Ne.prototype.has = ln, Ne.prototype.set = dn;
  function qe(t) {
    var r = -1, o = t ? t.length : 0;
    for (this.clear(); ++r < o; ) {
      var l = t[r];
      this.set(l[0], l[1]);
    }
  }
  function hn() {
    this.__data__ = [];
  }
  function pn(t) {
    var r = this.__data__, o = Tt(r, t);
    if (o < 0)
      return !1;
    var l = r.length - 1;
    return o == l ? r.pop() : on.call(r, o, 1), !0;
  }
  function We(t) {
    var r = this.__data__, o = Tt(r, t);
    return o < 0 ? void 0 : r[o][1];
  }
  function gn(t) {
    return Tt(this.__data__, t) > -1;
  }
  function _n(t, r) {
    var o = this.__data__, l = Tt(o, t);
    return l < 0 ? o.push([t, r]) : o[l][1] = r, this;
  }
  qe.prototype.clear = hn, qe.prototype.delete = pn, qe.prototype.get = We, qe.prototype.has = gn, qe.prototype.set = _n;
  function je(t) {
    var r = -1, o = t ? t.length : 0;
    for (this.clear(); ++r < o; ) {
      var l = t[r];
      this.set(l[0], l[1]);
    }
  }
  function yn() {
    this.__data__ = {
      hash: new Ne(),
      map: new (tt || qe)(),
      string: new Ne()
    };
  }
  function vn(t) {
    return gt(this, t).delete(t);
  }
  function ze(t) {
    return gt(this, t).get(t);
  }
  function mn(t) {
    return gt(this, t).has(t);
  }
  function wn(t, r) {
    return gt(this, t).set(t, r), this;
  }
  je.prototype.clear = yn, je.prototype.delete = vn, je.prototype.get = ze, je.prototype.has = mn, je.prototype.set = wn;
  function bt(t) {
    var r = -1, o = t ? t.length : 0;
    for (this.__data__ = new je(); ++r < o; )
      this.add(t[r]);
  }
  function bn(t) {
    return this.__data__.set(t, U), this;
  }
  function Tn(t) {
    return this.__data__.has(t);
  }
  bt.prototype.add = bt.prototype.push = bn, bt.prototype.has = Tn;
  function Re(t) {
    this.__data__ = new qe(t);
  }
  function En() {
    this.__data__ = new qe();
  }
  function An(t) {
    return this.__data__.delete(t);
  }
  function Je(t) {
    return this.__data__.get(t);
  }
  function On(t) {
    return this.__data__.has(t);
  }
  function xn(t, r) {
    var o = this.__data__;
    if (o instanceof qe) {
      var l = o.__data__;
      if (!tt || l.length < A - 1)
        return l.push([t, r]), this;
      o = this.__data__ = new je(l);
    }
    return o.set(t, r), this;
  }
  Re.prototype.clear = En, Re.prototype.delete = An, Re.prototype.get = Je, Re.prototype.has = On, Re.prototype.set = xn;
  function Cn(t, r) {
    var o = st(t) || Ht(t) ? Le(t.length, String) : [], l = o.length, I = !!l;
    for (var v in t)
      (r || Te.call(t, v)) && !(I && (v == "length" || Zt(v, l))) && o.push(v);
    return o;
  }
  function Tt(t, r) {
    for (var o = t.length; o--; )
      if (Bt(t[o][0], r))
        return o;
    return -1;
  }
  var Sn = Bn();
  function zt(t, r) {
    return t && Sn(t, r, Ve);
  }
  function pt(t, r) {
    r = Qe(r, t) ? [r] : Qt(r);
    for (var o = 0, l = r.length; t != null && o < l; )
      t = t[Ot(r[o++])];
    return o && o == l ? t : void 0;
  }
  function Jt(t) {
    return q.call(t);
  }
  function In(t, r) {
    return t != null && r in Object(t);
  }
  function Ut(t, r, o, l, I) {
    return t === r ? !0 : t == null || r == null || !yt(t) && !vt(r) ? t !== t && r !== r : Rn(t, r, Ut, o, l, I);
  }
  function Rn(t, r, o, l, I, v) {
    var K = st(t), J = st(r), te = d, e = d;
    K || (te = Xe(t), te = te == C ? pe : te), J || (e = Xe(r), e = e == C ? pe : e);
    var n = te == pe && !ee(t), s = e == pe && !ee(r), u = te == e;
    if (u && !n)
      return v || (v = new Re()), K || Kt(t) ? Xt(t, r, o, l, I, v) : Hn(t, r, te, o, l, I, v);
    if (!(I & L)) {
      var T = n && Te.call(t, "__wrapped__"), p = s && Te.call(r, "__wrapped__");
      if (T || p) {
        var M = T ? t.value() : t, k = p ? r.value() : r;
        return v || (v = new Re()), o(M, k, l, I, v);
      }
    }
    return u ? (v || (v = new Re()), Yt(t, r, o, l, I, v)) : !1;
  }
  function kt(t, r, o, l) {
    var I = o.length, v = I, K = !l;
    if (t == null)
      return !v;
    for (t = Object(t); I--; ) {
      var J = o[I];
      if (K && J[2] ? J[1] !== t[J[0]] : !(J[0] in t))
        return !1;
    }
    for (; ++I < v; ) {
      J = o[I];
      var te = J[0], e = t[te], n = J[1];
      if (K && J[2]) {
        if (e === void 0 && !(te in t))
          return !1;
      } else {
        var s = new Re();
        if (l)
          var u = l(e, n, te, t, r, s);
        if (!(u === void 0 ? Ut(n, e, l, W | L, s) : u))
          return !1;
      }
    }
    return !0;
  }
  function Pn(t) {
    if (!yt(t) || $n(t))
      return !1;
    var r = Ze(t) || ee(t) ? $e : se;
    return r.test(at(t));
  }
  function Ln(t) {
    return vt(t) && _t(t.length) && !!h[q.call(t)];
  }
  function Dn(t) {
    return typeof t == "function" ? t : t == null ? $t : typeof t == "object" ? st(t) ? Un(t[0], t[1]) : Fn(t) : Wn(t);
  }
  function Mn(t) {
    if (!Ye(t))
      return un(t);
    var r = [];
    for (var o in Object(t))
      Te.call(t, o) && o != "constructor" && r.push(o);
    return r;
  }
  function Fn(t) {
    var r = Kn(t);
    return r.length == 1 && r[0][2] ? Nt(r[0][0], r[0][1]) : function(o) {
      return o === t || kt(o, t, r);
    };
  }
  function Un(t, r) {
    return Qe(t) && Vt(r) ? Nt(Ot(t), r) : function(o) {
      var l = tn(o, t);
      return l === void 0 && l === r ? Gt(o, t) : Ut(r, l, void 0, W | L);
    };
  }
  function kn(t) {
    return function(r) {
      return pt(r, t);
    };
  }
  function Nn(t) {
    if (typeof t == "string")
      return t;
    if (Be(t))
      return Wt ? Wt.call(t) : "";
    var r = t + "";
    return r == "0" && 1 / t == -ue ? "-0" : r;
  }
  function Qt(t) {
    return st(t) ? t : At(t);
  }
  function Bn(t) {
    return function(r, o, l) {
      for (var I = -1, v = Object(r), K = l(r), J = K.length; J--; ) {
        var te = K[t ? J : ++I];
        if (o(v[te], te, v) === !1)
          break;
      }
      return r;
    };
  }
  function Xt(t, r, o, l, I, v) {
    var K = I & L, J = t.length, te = r.length;
    if (J != te && !(K && te > J))
      return !1;
    var e = v.get(t);
    if (e && v.get(r))
      return e == r;
    var n = -1, s = !0, u = I & W ? new bt() : void 0;
    for (v.set(t, r), v.set(r, t); ++n < J; ) {
      var T = t[n], p = r[n];
      if (l)
        var M = K ? l(p, T, n, r, t, v) : l(T, p, n, t, r, v);
      if (M !== void 0) {
        if (M)
          continue;
        s = !1;
        break;
      }
      if (u) {
        if (!we(r, function(k, Q) {
          if (!u.has(Q) && (T === k || o(T, k, l, I, v)))
            return u.add(Q);
        })) {
          s = !1;
          break;
        }
      } else if (!(T === p || o(T, p, l, I, v))) {
        s = !1;
        break;
      }
    }
    return v.delete(t), v.delete(r), s;
  }
  function Hn(t, r, o, l, I, v, K) {
    switch (o) {
      case f:
        if (t.byteLength != r.byteLength || t.byteOffset != r.byteOffset)
          return !1;
        t = t.buffer, r = r.buffer;
      case i:
        return !(t.byteLength != r.byteLength || !l(new It(t), new It(r)));
      case R:
      case P:
      case Ue:
        return Bt(+t, +r);
      case j:
        return t.name == r.name && t.message == r.message;
      case ut:
      case Se:
        return t == r + "";
      case he:
        var J = De;
      case Pe:
        var te = v & L;
        if (J || (J = ke), t.size != r.size && !te)
          return !1;
        var e = K.get(t);
        if (e)
          return e == r;
        v |= W, K.set(t, r);
        var n = Xt(J(t), J(r), l, I, v, K);
        return K.delete(t), n;
      case B:
        if (Mt)
          return Mt.call(t) == Mt.call(r);
    }
    return !1;
  }
  function Yt(t, r, o, l, I, v) {
    var K = I & L, J = Ve(t), te = J.length, e = Ve(r), n = e.length;
    if (te != n && !K)
      return !1;
    for (var s = te; s--; ) {
      var u = J[s];
      if (!(K ? u in r : Te.call(r, u)))
        return !1;
    }
    var T = v.get(t);
    if (T && v.get(r))
      return T == r;
    var p = !0;
    v.set(t, r), v.set(r, t);
    for (var M = K; ++s < te; ) {
      u = J[s];
      var k = t[u], Q = r[u];
      if (l)
        var oe = K ? l(Q, k, u, r, t, v) : l(k, Q, u, t, r, v);
      if (!(oe === void 0 ? k === Q || o(k, Q, l, I, v) : oe)) {
        p = !1;
        break;
      }
      M || (M = u == "constructor");
    }
    if (p && !M) {
      var de = t.constructor, ge = r.constructor;
      de != ge && "constructor" in t && "constructor" in r && !(typeof de == "function" && de instanceof de && typeof ge == "function" && ge instanceof ge) && (p = !1);
    }
    return v.delete(t), v.delete(r), p;
  }
  function gt(t, r) {
    var o = t.__data__;
    return Gn(r) ? o[typeof r == "string" ? "string" : "hash"] : o.map;
  }
  function Kn(t) {
    for (var r = Ve(t), o = r.length; o--; ) {
      var l = r[o], I = t[l];
      r[o] = [l, I, Vt(I)];
    }
    return r;
  }
  function ct(t, r) {
    var o = Y(t, r);
    return Pn(o) ? o : void 0;
  }
  var Xe = Jt;
  (Rt && Xe(new Rt(new ArrayBuffer(1))) != f || tt && Xe(new tt()) != he || Pt && Xe(Pt.resolve()) != Ce || dt && Xe(new dt()) != Pe || nt && Xe(new nt()) != a) && (Xe = function(t) {
    var r = q.call(t), o = r == pe ? t.constructor : void 0, l = o ? at(o) : void 0;
    if (l)
      switch (l) {
        case Lt:
          return f;
        case Dt:
          return he;
        case ht:
          return Ce;
        case fn:
          return Pe;
        case cn:
          return a;
      }
    return r;
  });
  function Et(t, r, o) {
    r = Qe(r, t) ? [r] : Qt(r);
    for (var l, I = -1, K = r.length; ++I < K; ) {
      var v = Ot(r[I]);
      if (!(l = t != null && o(t, v)))
        break;
      t = t[v];
    }
    if (l)
      return l;
    var K = t ? t.length : 0;
    return !!K && _t(K) && Zt(v, K) && (st(t) || Ht(t));
  }
  function Zt(t, r) {
    return r = r ?? E, !!r && (typeof t == "number" || ie.test(t)) && t > -1 && t % 1 == 0 && t < r;
  }
  function Qe(t, r) {
    if (st(t))
      return !1;
    var o = typeof t;
    return o == "number" || o == "symbol" || o == "boolean" || t == null || Be(t) ? !0 : fe.test(t) || !Ae.test(t) || r != null && t in Object(r);
  }
  function Gn(t) {
    var r = typeof t;
    return r == "string" || r == "number" || r == "symbol" || r == "boolean" ? t !== "__proto__" : t === null;
  }
  function $n(t) {
    return !!xe && xe in t;
  }
  function Ye(t) {
    var r = t && t.constructor, o = typeof r == "function" && r.prototype || Oe;
    return t === o;
  }
  function Vt(t) {
    return t === t && !yt(t);
  }
  function Nt(t, r) {
    return function(o) {
      return o == null ? !1 : o[t] === r && (r !== void 0 || t in Object(o));
    };
  }
  var At = xt(function(t) {
    t = qn(t);
    var r = [];
    return y.test(t) && r.push(""), t.replace(z, function(o, l, I, v) {
      r.push(I ? v.replace(V, "$1") : l || o);
    }), r;
  });
  function Ot(t) {
    if (typeof t == "string" || Be(t))
      return t;
    var r = t + "";
    return r == "0" && 1 / t == -ue ? "-0" : r;
  }
  function at(t) {
    if (t != null) {
      try {
        return _e.call(t);
      } catch {
      }
      try {
        return t + "";
      } catch {
      }
    }
    return "";
  }
  function xt(t, r) {
    if (typeof t != "function" || r && typeof r != "function")
      throw new TypeError(g);
    var o = function() {
      var l = arguments, I = r ? r.apply(this, l) : l[0], v = o.cache;
      if (v.has(I))
        return v.get(I);
      var K = t.apply(this, l);
      return o.cache = v.set(I, K), K;
    };
    return o.cache = new (xt.Cache || je)(), o;
  }
  xt.Cache = je;
  function Bt(t, r) {
    return t === r || t !== t && r !== r;
  }
  function Ht(t) {
    return Ct(t) && Te.call(t, "callee") && (!qt.call(t, "callee") || q.call(t) == C);
  }
  var st = Array.isArray;
  function en(t) {
    return t != null && _t(t.length) && !Ze(t);
  }
  function Ct(t) {
    return vt(t) && en(t);
  }
  function Ze(t) {
    var r = yt(t) ? q.call(t) : "";
    return r == Fe || r == et;
  }
  function _t(t) {
    return typeof t == "number" && t > -1 && t % 1 == 0 && t <= E;
  }
  function yt(t) {
    var r = typeof t;
    return !!t && (r == "object" || r == "function");
  }
  function vt(t) {
    return !!t && typeof t == "object";
  }
  function Be(t) {
    return typeof t == "symbol" || vt(t) && q.call(t) == B;
  }
  var Kt = me ? Ee(me) : Ln;
  function qn(t) {
    return t == null ? "" : Nn(t);
  }
  function tn(t, r, o) {
    var l = t == null ? void 0 : pt(t, r);
    return l === void 0 ? o : l;
  }
  function Gt(t, r) {
    return t != null && Et(t, r, In);
  }
  function Ve(t) {
    return en(t) ? Cn(t) : Mn(t);
  }
  function St(t, r) {
    var o = {};
    return r = Dn(r), zt(t, function(l, I, v) {
      o[r(l, I, v)] = l;
    }), o;
  }
  function $t(t) {
    return t;
  }
  function Wn(t) {
    return Qe(t) ? Ie(Ot(t)) : kn(t);
  }
  c.exports = St;
})(an, an.exports);
var lr = an.exports;
const Xn = /* @__PURE__ */ jn(lr);
var sn = { exports: {} };
sn.exports;
(function(c, O) {
  var A = 200, g = "Expected a function", U = "__lodash_hash_undefined__", W = 1, L = 2, ue = 1 / 0, E = 9007199254740991, C = "[object Arguments]", d = "[object Array]", R = "[object Boolean]", P = "[object Date]", j = "[object Error]", Fe = "[object Function]", et = "[object GeneratorFunction]", he = "[object Map]", Ue = "[object Number]", pe = "[object Object]", Ce = "[object Promise]", ut = "[object RegExp]", Pe = "[object Set]", Se = "[object String]", B = "[object Symbol]", a = "[object WeakMap]", i = "[object ArrayBuffer]", f = "[object DataView]", _ = "[object Float32Array]", m = "[object Float64Array]", S = "[object Int8Array]", F = "[object Int16Array]", ae = "[object Int32Array]", Z = "[object Uint8Array]", w = "[object Uint8ClampedArray]", ne = "[object Uint16Array]", x = "[object Uint32Array]", Ae = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, fe = /^\w*$/, y = /^\./, z = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, ce = /[\\^$.*+?()[\]{}|]/g, V = /\\(\\)?/g, se = /^\[object .+?Constructor\]$/, ie = /^(?:0|[1-9]\d*)$/, h = {};
  h[_] = h[m] = h[S] = h[F] = h[ae] = h[Z] = h[w] = h[ne] = h[x] = !0, h[C] = h[d] = h[i] = h[R] = h[f] = h[P] = h[j] = h[Fe] = h[he] = h[Ue] = h[pe] = h[ut] = h[Pe] = h[Se] = h[a] = !1;
  var b = typeof ve == "object" && ve && ve.Object === Object && ve, N = typeof self == "object" && self && self.Object === Object && self, D = b || N || Function("return this")(), re = O && !O.nodeType && O, H = re && !0 && c && !c.nodeType && c, le = H && H.exports === re, X = le && b.process, G = function() {
    try {
      return X && X.binding("util");
    } catch {
    }
  }(), me = G && G.isTypedArray;
  function we(e, n) {
    for (var s = -1, u = n.length, T = e.length; ++s < u; )
      e[T + s] = n[s];
    return e;
  }
  function Ie(e, n) {
    for (var s = -1, u = e ? e.length : 0; ++s < u; )
      if (n(e[s], s, e))
        return !0;
    return !1;
  }
  function Le(e) {
    return function(n) {
      return n == null ? void 0 : n[e];
    };
  }
  function Ee(e, n) {
    for (var s = -1, u = Array(e); ++s < e; )
      u[s] = n(s);
    return u;
  }
  function Y(e) {
    return function(n) {
      return e(n);
    };
  }
  function ee(e, n) {
    return e == null ? void 0 : e[n];
  }
  function De(e) {
    var n = !1;
    if (e != null && typeof e.toString != "function")
      try {
        n = !!(e + "");
      } catch {
      }
    return n;
  }
  function $(e) {
    var n = -1, s = Array(e.size);
    return e.forEach(function(u, T) {
      s[++n] = [T, u];
    }), s;
  }
  function ke(e, n) {
    return function(s) {
      return e(n(s));
    };
  }
  function Me(e) {
    var n = -1, s = Array(e.size);
    return e.forEach(function(u) {
      s[++n] = u;
    }), s;
  }
  var Ge = Array.prototype, Oe = Function.prototype, be = Object.prototype, xe = D["__core-js_shared__"], _e = function() {
    var e = /[^.]+$/.exec(xe && xe.keys && xe.keys.IE_PROTO || "");
    return e ? "Symbol(src)_1." + e : "";
  }(), Te = Oe.toString, q = be.hasOwnProperty, $e = be.toString, ft = RegExp(
    "^" + Te.call(q).replace(ce, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
  ), It = D.Symbol, qt = D.Uint8Array, on = ke(Object.getPrototypeOf, Object), un = be.propertyIsEnumerable, Rt = Ge.splice, tt = Object.getOwnPropertySymbols, Pt = ke(Object.keys, Object), dt = Qe(D, "DataView"), nt = Qe(D, "Map"), rt = Qe(D, "Promise"), Lt = Qe(D, "Set"), Dt = Qe(D, "WeakMap"), ht = Qe(Object, "create"), fn = Ze(dt), cn = Ze(nt), wt = Ze(rt), Mt = Ze(Lt), Wt = Ze(Dt), Ne = It ? It.prototype : void 0, Ft = Ne ? Ne.valueOf : void 0, jt = Ne ? Ne.toString : void 0;
  function it(e) {
    var n = -1, s = e ? e.length : 0;
    for (this.clear(); ++n < s; ) {
      var u = e[n];
      this.set(u[0], u[1]);
    }
  }
  function ln() {
    this.__data__ = ht ? ht(null) : {};
  }
  function dn(e) {
    return this.has(e) && delete this.__data__[e];
  }
  function qe(e) {
    var n = this.__data__;
    if (ht) {
      var s = n[e];
      return s === U ? void 0 : s;
    }
    return q.call(n, e) ? n[e] : void 0;
  }
  function hn(e) {
    var n = this.__data__;
    return ht ? n[e] !== void 0 : q.call(n, e);
  }
  function pn(e, n) {
    var s = this.__data__;
    return s[e] = ht && n === void 0 ? U : n, this;
  }
  it.prototype.clear = ln, it.prototype.delete = dn, it.prototype.get = qe, it.prototype.has = hn, it.prototype.set = pn;
  function We(e) {
    var n = -1, s = e ? e.length : 0;
    for (this.clear(); ++n < s; ) {
      var u = e[n];
      this.set(u[0], u[1]);
    }
  }
  function gn() {
    this.__data__ = [];
  }
  function _n(e) {
    var n = this.__data__, s = pt(n, e);
    if (s < 0)
      return !1;
    var u = n.length - 1;
    return s == u ? n.pop() : Rt.call(n, s, 1), !0;
  }
  function je(e) {
    var n = this.__data__, s = pt(n, e);
    return s < 0 ? void 0 : n[s][1];
  }
  function yn(e) {
    return pt(this.__data__, e) > -1;
  }
  function vn(e, n) {
    var s = this.__data__, u = pt(s, e);
    return u < 0 ? s.push([e, n]) : s[u][1] = n, this;
  }
  We.prototype.clear = gn, We.prototype.delete = _n, We.prototype.get = je, We.prototype.has = yn, We.prototype.set = vn;
  function ze(e) {
    var n = -1, s = e ? e.length : 0;
    for (this.clear(); ++n < s; ) {
      var u = e[n];
      this.set(u[0], u[1]);
    }
  }
  function mn() {
    this.__data__ = {
      hash: new it(),
      map: new (nt || We)(),
      string: new it()
    };
  }
  function wn(e) {
    return Et(this, e).delete(e);
  }
  function bt(e) {
    return Et(this, e).get(e);
  }
  function bn(e) {
    return Et(this, e).has(e);
  }
  function Tn(e, n) {
    return Et(this, e).set(e, n), this;
  }
  ze.prototype.clear = mn, ze.prototype.delete = wn, ze.prototype.get = bt, ze.prototype.has = bn, ze.prototype.set = Tn;
  function Re(e) {
    var n = -1, s = e ? e.length : 0;
    for (this.__data__ = new ze(); ++n < s; )
      this.add(e[n]);
  }
  function En(e) {
    return this.__data__.set(e, U), this;
  }
  function An(e) {
    return this.__data__.has(e);
  }
  Re.prototype.add = Re.prototype.push = En, Re.prototype.has = An;
  function Je(e) {
    this.__data__ = new We(e);
  }
  function On() {
    this.__data__ = new We();
  }
  function xn(e) {
    return this.__data__.delete(e);
  }
  function Cn(e) {
    return this.__data__.get(e);
  }
  function Tt(e) {
    return this.__data__.has(e);
  }
  function Sn(e, n) {
    var s = this.__data__;
    if (s instanceof We) {
      var u = s.__data__;
      if (!nt || u.length < A - 1)
        return u.push([e, n]), this;
      s = this.__data__ = new ze(u);
    }
    return s.set(e, n), this;
  }
  Je.prototype.clear = On, Je.prototype.delete = xn, Je.prototype.get = Cn, Je.prototype.has = Tt, Je.prototype.set = Sn;
  function zt(e, n) {
    var s = Be(e) || vt(e) ? Ee(e.length, String) : [], u = s.length, T = !!u;
    for (var p in e)
      (n || q.call(e, p)) && !(T && (p == "length" || Nt(p, u))) && s.push(p);
    return s;
  }
  function pt(e, n) {
    for (var s = e.length; s--; )
      if (yt(e[s][0], n))
        return s;
    return -1;
  }
  function Jt(e, n) {
    n = At(n, e) ? [n] : Yt(n);
    for (var s = 0, u = n.length; e != null && s < u; )
      e = e[Ct(n[s++])];
    return s && s == u ? e : void 0;
  }
  function In(e, n, s) {
    var u = n(e);
    return Be(e) ? u : we(u, s(e));
  }
  function Ut(e) {
    return $e.call(e);
  }
  function Rn(e, n) {
    return e != null && n in Object(e);
  }
  function kt(e, n, s, u, T) {
    return e === n ? !0 : e == null || n == null || !Ve(e) && !St(n) ? e !== e && n !== n : Pn(e, n, kt, s, u, T);
  }
  function Pn(e, n, s, u, T, p) {
    var M = Be(e), k = Be(n), Q = d, oe = d;
    M || (Q = Ye(e), Q = Q == C ? pe : Q), k || (oe = Ye(n), oe = oe == C ? pe : oe);
    var de = Q == pe && !De(e), ge = oe == pe && !De(n), ye = Q == oe;
    if (ye && !de)
      return p || (p = new Je()), M || Wn(e) ? gt(e, n, s, u, T, p) : Kn(e, n, Q, s, u, T, p);
    if (!(T & L)) {
      var He = de && q.call(e, "__wrapped__"), Ke = ge && q.call(n, "__wrapped__");
      if (He || Ke) {
        var lt = He ? e.value() : e, ot = Ke ? n.value() : n;
        return p || (p = new Je()), s(lt, ot, u, T, p);
      }
    }
    return ye ? (p || (p = new Je()), ct(e, n, s, u, T, p)) : !1;
  }
  function Ln(e, n, s, u) {
    var T = s.length, p = T, M = !u;
    if (e == null)
      return !p;
    for (e = Object(e); T--; ) {
      var k = s[T];
      if (M && k[2] ? k[1] !== e[k[0]] : !(k[0] in e))
        return !1;
    }
    for (; ++T < p; ) {
      k = s[T];
      var Q = k[0], oe = e[Q], de = k[1];
      if (M && k[2]) {
        if (oe === void 0 && !(Q in e))
          return !1;
      } else {
        var ge = new Je();
        if (u)
          var ye = u(oe, de, Q, e, n, ge);
        if (!(ye === void 0 ? kt(de, oe, u, W | L, ge) : ye))
          return !1;
      }
    }
    return !0;
  }
  function Dn(e) {
    if (!Ve(e) || at(e))
      return !1;
    var n = tn(e) || De(e) ? ft : se;
    return n.test(Ze(e));
  }
  function Mn(e) {
    return St(e) && Gt(e.length) && !!h[$e.call(e)];
  }
  function Fn(e) {
    return typeof e == "function" ? e : e == null ? K : typeof e == "object" ? Be(e) ? Qt(e[0], e[1]) : Nn(e) : J(e);
  }
  function Un(e) {
    if (!xt(e))
      return Pt(e);
    var n = [];
    for (var s in Object(e))
      q.call(e, s) && s != "constructor" && n.push(s);
    return n;
  }
  function kn(e) {
    if (!Ve(e))
      return st(e);
    var n = xt(e), s = [];
    for (var u in e)
      u == "constructor" && (n || !q.call(e, u)) || s.push(u);
    return s;
  }
  function Nn(e) {
    var n = Zt(e);
    return n.length == 1 && n[0][2] ? Ht(n[0][0], n[0][1]) : function(s) {
      return s === e || Ln(s, e, n);
    };
  }
  function Qt(e, n) {
    return At(e) && Bt(n) ? Ht(Ct(e), n) : function(s) {
      var u = r(s, e);
      return u === void 0 && u === n ? o(s, e) : kt(n, u, void 0, W | L);
    };
  }
  function Bn(e, n, s) {
    for (var u = -1, T = n.length, p = {}; ++u < T; ) {
      var M = n[u], k = e[M];
      s(k, M) && (p[M] = k);
    }
    return p;
  }
  function Xt(e) {
    return function(n) {
      return Jt(n, e);
    };
  }
  function Hn(e) {
    if (typeof e == "string")
      return e;
    if ($t(e))
      return jt ? jt.call(e) : "";
    var n = e + "";
    return n == "0" && 1 / e == -ue ? "-0" : n;
  }
  function Yt(e) {
    return Be(e) ? e : en(e);
  }
  function gt(e, n, s, u, T, p) {
    var M = T & L, k = e.length, Q = n.length;
    if (k != Q && !(M && Q > k))
      return !1;
    var oe = p.get(e);
    if (oe && p.get(n))
      return oe == n;
    var de = -1, ge = !0, ye = T & W ? new Re() : void 0;
    for (p.set(e, n), p.set(n, e); ++de < k; ) {
      var He = e[de], Ke = n[de];
      if (u)
        var lt = M ? u(Ke, He, de, n, e, p) : u(He, Ke, de, e, n, p);
      if (lt !== void 0) {
        if (lt)
          continue;
        ge = !1;
        break;
      }
      if (ye) {
        if (!Ie(n, function(ot, mt) {
          if (!ye.has(mt) && (He === ot || s(He, ot, u, T, p)))
            return ye.add(mt);
        })) {
          ge = !1;
          break;
        }
      } else if (!(He === Ke || s(He, Ke, u, T, p))) {
        ge = !1;
        break;
      }
    }
    return p.delete(e), p.delete(n), ge;
  }
  function Kn(e, n, s, u, T, p, M) {
    switch (s) {
      case f:
        if (e.byteLength != n.byteLength || e.byteOffset != n.byteOffset)
          return !1;
        e = e.buffer, n = n.buffer;
      case i:
        return !(e.byteLength != n.byteLength || !u(new qt(e), new qt(n)));
      case R:
      case P:
      case Ue:
        return yt(+e, +n);
      case j:
        return e.name == n.name && e.message == n.message;
      case ut:
      case Se:
        return e == n + "";
      case he:
        var k = $;
      case Pe:
        var Q = p & L;
        if (k || (k = Me), e.size != n.size && !Q)
          return !1;
        var oe = M.get(e);
        if (oe)
          return oe == n;
        p |= W, M.set(e, n);
        var de = gt(k(e), k(n), u, T, p, M);
        return M.delete(e), de;
      case B:
        if (Ft)
          return Ft.call(e) == Ft.call(n);
    }
    return !1;
  }
  function ct(e, n, s, u, T, p) {
    var M = T & L, k = l(e), Q = k.length, oe = l(n), de = oe.length;
    if (Q != de && !M)
      return !1;
    for (var ge = Q; ge--; ) {
      var ye = k[ge];
      if (!(M ? ye in n : q.call(n, ye)))
        return !1;
    }
    var He = p.get(e);
    if (He && p.get(n))
      return He == n;
    var Ke = !0;
    p.set(e, n), p.set(n, e);
    for (var lt = M; ++ge < Q; ) {
      ye = k[ge];
      var ot = e[ye], mt = n[ye];
      if (u)
        var Qn = M ? u(mt, ot, ye, n, e, p) : u(ot, mt, ye, e, n, p);
      if (!(Qn === void 0 ? ot === mt || s(ot, mt, u, T, p) : Qn)) {
        Ke = !1;
        break;
      }
      lt || (lt = ye == "constructor");
    }
    if (Ke && !lt) {
      var nn = e.constructor, rn = n.constructor;
      nn != rn && "constructor" in e && "constructor" in n && !(typeof nn == "function" && nn instanceof nn && typeof rn == "function" && rn instanceof rn) && (Ke = !1);
    }
    return p.delete(e), p.delete(n), Ke;
  }
  function Xe(e) {
    return In(e, I, $n);
  }
  function Et(e, n) {
    var s = e.__data__;
    return Ot(n) ? s[typeof n == "string" ? "string" : "hash"] : s.map;
  }
  function Zt(e) {
    for (var n = l(e), s = n.length; s--; ) {
      var u = n[s], T = e[u];
      n[s] = [u, T, Bt(T)];
    }
    return n;
  }
  function Qe(e, n) {
    var s = ee(e, n);
    return Dn(s) ? s : void 0;
  }
  var Gn = tt ? ke(tt, Object) : te, $n = tt ? function(e) {
    for (var n = []; e; )
      we(n, Gn(e)), e = on(e);
    return n;
  } : te, Ye = Ut;
  (dt && Ye(new dt(new ArrayBuffer(1))) != f || nt && Ye(new nt()) != he || rt && Ye(rt.resolve()) != Ce || Lt && Ye(new Lt()) != Pe || Dt && Ye(new Dt()) != a) && (Ye = function(e) {
    var n = $e.call(e), s = n == pe ? e.constructor : void 0, u = s ? Ze(s) : void 0;
    if (u)
      switch (u) {
        case fn:
          return f;
        case cn:
          return he;
        case wt:
          return Ce;
        case Mt:
          return Pe;
        case Wt:
          return a;
      }
    return n;
  });
  function Vt(e, n, s) {
    n = At(n, e) ? [n] : Yt(n);
    for (var u, T = -1, M = n.length; ++T < M; ) {
      var p = Ct(n[T]);
      if (!(u = e != null && s(e, p)))
        break;
      e = e[p];
    }
    if (u)
      return u;
    var M = e ? e.length : 0;
    return !!M && Gt(M) && Nt(p, M) && (Be(e) || vt(e));
  }
  function Nt(e, n) {
    return n = n ?? E, !!n && (typeof e == "number" || ie.test(e)) && e > -1 && e % 1 == 0 && e < n;
  }
  function At(e, n) {
    if (Be(e))
      return !1;
    var s = typeof e;
    return s == "number" || s == "symbol" || s == "boolean" || e == null || $t(e) ? !0 : fe.test(e) || !Ae.test(e) || n != null && e in Object(n);
  }
  function Ot(e) {
    var n = typeof e;
    return n == "string" || n == "number" || n == "symbol" || n == "boolean" ? e !== "__proto__" : e === null;
  }
  function at(e) {
    return !!_e && _e in e;
  }
  function xt(e) {
    var n = e && e.constructor, s = typeof n == "function" && n.prototype || be;
    return e === s;
  }
  function Bt(e) {
    return e === e && !Ve(e);
  }
  function Ht(e, n) {
    return function(s) {
      return s == null ? !1 : s[e] === n && (n !== void 0 || e in Object(s));
    };
  }
  function st(e) {
    var n = [];
    if (e != null)
      for (var s in Object(e))
        n.push(s);
    return n;
  }
  var en = _t(function(e) {
    e = t(e);
    var n = [];
    return y.test(e) && n.push(""), e.replace(z, function(s, u, T, p) {
      n.push(T ? p.replace(V, "$1") : u || s);
    }), n;
  });
  function Ct(e) {
    if (typeof e == "string" || $t(e))
      return e;
    var n = e + "";
    return n == "0" && 1 / e == -ue ? "-0" : n;
  }
  function Ze(e) {
    if (e != null) {
      try {
        return Te.call(e);
      } catch {
      }
      try {
        return e + "";
      } catch {
      }
    }
    return "";
  }
  function _t(e, n) {
    if (typeof e != "function" || n && typeof n != "function")
      throw new TypeError(g);
    var s = function() {
      var u = arguments, T = n ? n.apply(this, u) : u[0], p = s.cache;
      if (p.has(T))
        return p.get(T);
      var M = e.apply(this, u);
      return s.cache = p.set(T, M), M;
    };
    return s.cache = new (_t.Cache || ze)(), s;
  }
  _t.Cache = ze;
  function yt(e, n) {
    return e === n || e !== e && n !== n;
  }
  function vt(e) {
    return qn(e) && q.call(e, "callee") && (!un.call(e, "callee") || $e.call(e) == C);
  }
  var Be = Array.isArray;
  function Kt(e) {
    return e != null && Gt(e.length) && !tn(e);
  }
  function qn(e) {
    return St(e) && Kt(e);
  }
  function tn(e) {
    var n = Ve(e) ? $e.call(e) : "";
    return n == Fe || n == et;
  }
  function Gt(e) {
    return typeof e == "number" && e > -1 && e % 1 == 0 && e <= E;
  }
  function Ve(e) {
    var n = typeof e;
    return !!e && (n == "object" || n == "function");
  }
  function St(e) {
    return !!e && typeof e == "object";
  }
  function $t(e) {
    return typeof e == "symbol" || St(e) && $e.call(e) == B;
  }
  var Wn = me ? Y(me) : Mn;
  function t(e) {
    return e == null ? "" : Hn(e);
  }
  function r(e, n, s) {
    var u = e == null ? void 0 : Jt(e, n);
    return u === void 0 ? s : u;
  }
  function o(e, n) {
    return e != null && Vt(e, n, Rn);
  }
  function l(e) {
    return Kt(e) ? zt(e) : Un(e);
  }
  function I(e) {
    return Kt(e) ? zt(e, !0) : kn(e);
  }
  function v(e, n) {
    return e == null ? {} : Bn(e, Xe(e), Fn(n));
  }
  function K(e) {
    return e;
  }
  function J(e) {
    return At(e) ? Le(Ct(e)) : Xt(e);
  }
  function te() {
    return [];
  }
  c.exports = v;
})(sn, sn.exports);
var dr = sn.exports;
const hr = /* @__PURE__ */ jn(dr);
var pr = 1 / 0, gr = 9007199254740991, _r = "[object Arguments]", yr = "[object Function]", vr = "[object GeneratorFunction]", mr = "[object Symbol]", wr = typeof ve == "object" && ve && ve.Object === Object && ve, br = typeof self == "object" && self && self.Object === Object && self, Tr = wr || br || Function("return this")();
function Er(c, O, A) {
  switch (A.length) {
    case 0:
      return c.call(O);
    case 1:
      return c.call(O, A[0]);
    case 2:
      return c.call(O, A[0], A[1]);
    case 3:
      return c.call(O, A[0], A[1], A[2]);
  }
  return c.apply(O, A);
}
function Ar(c, O) {
  for (var A = -1, g = c ? c.length : 0, U = Array(g); ++A < g; )
    U[A] = O(c[A], A, c);
  return U;
}
function Or(c, O) {
  for (var A = -1, g = O.length, U = c.length; ++A < g; )
    c[U + A] = O[A];
  return c;
}
var zn = Object.prototype, xr = zn.hasOwnProperty, Jn = zn.toString, Yn = Tr.Symbol, Cr = zn.propertyIsEnumerable, Zn = Yn ? Yn.isConcatSpreadable : void 0, Vn = Math.max;
function er(c, O, A, g, U) {
  var W = -1, L = c.length;
  for (A || (A = Pr), U || (U = []); ++W < L; ) {
    var ue = c[W];
    O > 0 && A(ue) ? O > 1 ? er(ue, O - 1, A, g, U) : Or(U, ue) : g || (U[U.length] = ue);
  }
  return U;
}
function Sr(c, O) {
  return c = Object(c), Ir(c, O, function(A, g) {
    return g in c;
  });
}
function Ir(c, O, A) {
  for (var g = -1, U = O.length, W = {}; ++g < U; ) {
    var L = O[g], ue = c[L];
    A(ue, L) && (W[L] = ue);
  }
  return W;
}
function Rr(c, O) {
  return O = Vn(O === void 0 ? c.length - 1 : O, 0), function() {
    for (var A = arguments, g = -1, U = Vn(A.length - O, 0), W = Array(U); ++g < U; )
      W[g] = A[O + g];
    g = -1;
    for (var L = Array(O + 1); ++g < O; )
      L[g] = A[g];
    return L[O] = W, Er(c, this, L);
  };
}
function Pr(c) {
  return Mr(c) || Dr(c) || !!(Zn && c && c[Zn]);
}
function Lr(c) {
  if (typeof c == "string" || Hr(c))
    return c;
  var O = c + "";
  return O == "0" && 1 / c == -pr ? "-0" : O;
}
function Dr(c) {
  return Ur(c) && xr.call(c, "callee") && (!Cr.call(c, "callee") || Jn.call(c) == _r);
}
var Mr = Array.isArray;
function Fr(c) {
  return c != null && Nr(c.length) && !kr(c);
}
function Ur(c) {
  return tr(c) && Fr(c);
}
function kr(c) {
  var O = Br(c) ? Jn.call(c) : "";
  return O == yr || O == vr;
}
function Nr(c) {
  return typeof c == "number" && c > -1 && c % 1 == 0 && c <= gr;
}
function Br(c) {
  var O = typeof c;
  return !!c && (O == "object" || O == "function");
}
function tr(c) {
  return !!c && typeof c == "object";
}
function Hr(c) {
  return typeof c == "symbol" || tr(c) && Jn.call(c) == mr;
}
var Kr = Rr(function(c, O) {
  return c == null ? {} : Sr(c, Ar(er(O, 1), Lr));
}), Gr = Kr;
const $r = /* @__PURE__ */ jn(Gr);
var nr = { exports: {} };
(function(c, O) {
  (function(A, g) {
    g();
  })(ve, function() {
    function A(C, d) {
      return typeof d > "u" ? d = { autoBom: !1 } : typeof d != "object" && (console.warn("Deprecated: Expected third argument to be a object"), d = { autoBom: !d }), d.autoBom && /^\s*(?:text\/\S*|application\/xml|\S*\/\S*\+xml)\s*;.*charset\s*=\s*utf-8/i.test(C.type) ? new Blob(["\uFEFF", C], { type: C.type }) : C;
    }
    function g(C, d, R) {
      var P = new XMLHttpRequest();
      P.open("GET", C), P.responseType = "blob", P.onload = function() {
        E(P.response, d, R);
      }, P.onerror = function() {
        console.error("could not download file");
      }, P.send();
    }
    function U(C) {
      var d = new XMLHttpRequest();
      d.open("HEAD", C, !1);
      try {
        d.send();
      } catch {
      }
      return 200 <= d.status && 299 >= d.status;
    }
    function W(C) {
      try {
        C.dispatchEvent(new MouseEvent("click"));
      } catch {
        var d = document.createEvent("MouseEvents");
        d.initMouseEvent("click", !0, !0, window, 0, 0, 0, 80, 20, !1, !1, !1, !1, 0, null), C.dispatchEvent(d);
      }
    }
    var L = typeof window == "object" && window.window === window ? window : typeof self == "object" && self.self === self ? self : typeof ve == "object" && ve.global === ve ? ve : void 0, ue = L.navigator && /Macintosh/.test(navigator.userAgent) && /AppleWebKit/.test(navigator.userAgent) && !/Safari/.test(navigator.userAgent), E = L.saveAs || (typeof window != "object" || window !== L ? function() {
    } : "download" in HTMLAnchorElement.prototype && !ue ? function(C, d, R) {
      var P = L.URL || L.webkitURL, j = document.createElement("a");
      d = d || C.name || "download", j.download = d, j.rel = "noopener", typeof C == "string" ? (j.href = C, j.origin === location.origin ? W(j) : U(j.href) ? g(C, d, R) : W(j, j.target = "_blank")) : (j.href = P.createObjectURL(C), setTimeout(function() {
        P.revokeObjectURL(j.href);
      }, 4e4), setTimeout(function() {
        W(j);
      }, 0));
    } : "msSaveOrOpenBlob" in navigator ? function(C, d, R) {
      if (d = d || C.name || "download", typeof C != "string")
        navigator.msSaveOrOpenBlob(A(C, R), d);
      else if (U(C))
        g(C, d, R);
      else {
        var P = document.createElement("a");
        P.href = C, P.target = "_blank", setTimeout(function() {
          W(P);
        });
      }
    } : function(C, d, R, P) {
      if (P = P || open("", "_blank"), P && (P.document.title = P.document.body.innerText = "downloading..."), typeof C == "string")
        return g(C, d, R);
      var j = C.type === "application/octet-stream", Fe = /constructor/i.test(L.HTMLElement) || L.safari, et = /CriOS\/[\d]+/.test(navigator.userAgent);
      if ((et || j && Fe || ue) && typeof FileReader < "u") {
        var he = new FileReader();
        he.onloadend = function() {
          var Ce = he.result;
          Ce = et ? Ce : Ce.replace(/^data:[^;]*;/, "data:attachment/file;"), P ? P.location.href = Ce : location = Ce, P = null;
        }, he.readAsDataURL(C);
      } else {
        var Ue = L.URL || L.webkitURL, pe = Ue.createObjectURL(C);
        P ? P.location = pe : location.href = pe, P = null, setTimeout(function() {
          Ue.revokeObjectURL(pe);
        }, 4e4);
      }
    });
    L.saveAs = E.saveAs = E, c.exports = E;
  });
})(nr);
var qr = nr.exports, rr = { exports: {} };
/* @license
Papa Parse
v5.4.1
https://github.com/mholt/PapaParse
License: MIT
*/
(function(c, O) {
  (function(A, g) {
    c.exports = g();
  })(ve, function A() {
    var g = typeof self < "u" ? self : typeof window < "u" ? window : g !== void 0 ? g : {}, U = !g.document && !!g.postMessage, W = g.IS_PAPA_WORKER || !1, L = {}, ue = 0, E = { parse: function(a, i) {
      var f = (i = i || {}).dynamicTyping || !1;
      if (B(f) && (i.dynamicTypingFunction = f, f = {}), i.dynamicTyping = f, i.transform = !!B(i.transform) && i.transform, i.worker && E.WORKERS_SUPPORTED) {
        var _ = function() {
          if (!E.WORKERS_SUPPORTED)
            return !1;
          var S = (ae = g.URL || g.webkitURL || null, Z = A.toString(), E.BLOB_URL || (E.BLOB_URL = ae.createObjectURL(new Blob(["var global = (function() { if (typeof self !== 'undefined') { return self; } if (typeof window !== 'undefined') { return window; } if (typeof global !== 'undefined') { return global; } return {}; })(); global.IS_PAPA_WORKER=true; ", "(", Z, ")();"], { type: "text/javascript" })))), F = new g.Worker(S), ae, Z;
          return F.onmessage = pe, F.id = ue++, L[F.id] = F;
        }();
        return _.userStep = i.step, _.userChunk = i.chunk, _.userComplete = i.complete, _.userError = i.error, i.step = B(i.step), i.chunk = B(i.chunk), i.complete = B(i.complete), i.error = B(i.error), delete i.worker, void _.postMessage({ input: a, config: i, workerId: _.id });
      }
      var m = null;
      return E.NODE_STREAM_INPUT, typeof a == "string" ? (a = function(S) {
        return S.charCodeAt(0) === 65279 ? S.slice(1) : S;
      }(a), m = i.download ? new R(i) : new j(i)) : a.readable === !0 && B(a.read) && B(a.on) ? m = new Fe(i) : (g.File && a instanceof File || a instanceof Object) && (m = new P(i)), m.stream(a);
    }, unparse: function(a, i) {
      var f = !1, _ = !0, m = ",", S = `\r
`, F = '"', ae = F + F, Z = !1, w = null, ne = !1;
      (function() {
        if (typeof i == "object") {
          if (typeof i.delimiter != "string" || E.BAD_DELIMITERS.filter(function(y) {
            return i.delimiter.indexOf(y) !== -1;
          }).length || (m = i.delimiter), (typeof i.quotes == "boolean" || typeof i.quotes == "function" || Array.isArray(i.quotes)) && (f = i.quotes), typeof i.skipEmptyLines != "boolean" && typeof i.skipEmptyLines != "string" || (Z = i.skipEmptyLines), typeof i.newline == "string" && (S = i.newline), typeof i.quoteChar == "string" && (F = i.quoteChar), typeof i.header == "boolean" && (_ = i.header), Array.isArray(i.columns)) {
            if (i.columns.length === 0)
              throw new Error("Option columns is empty");
            w = i.columns;
          }
          i.escapeChar !== void 0 && (ae = i.escapeChar + F), (typeof i.escapeFormulae == "boolean" || i.escapeFormulae instanceof RegExp) && (ne = i.escapeFormulae instanceof RegExp ? i.escapeFormulae : /^[=+\-@\t\r].*$/);
        }
      })();
      var x = new RegExp(he(F), "g");
      if (typeof a == "string" && (a = JSON.parse(a)), Array.isArray(a)) {
        if (!a.length || Array.isArray(a[0]))
          return Ae(null, a, Z);
        if (typeof a[0] == "object")
          return Ae(w || Object.keys(a[0]), a, Z);
      } else if (typeof a == "object")
        return typeof a.data == "string" && (a.data = JSON.parse(a.data)), Array.isArray(a.data) && (a.fields || (a.fields = a.meta && a.meta.fields || w), a.fields || (a.fields = Array.isArray(a.data[0]) ? a.fields : typeof a.data[0] == "object" ? Object.keys(a.data[0]) : []), Array.isArray(a.data[0]) || typeof a.data[0] == "object" || (a.data = [a.data])), Ae(a.fields || [], a.data || [], Z);
      throw new Error("Unable to serialize unrecognized input");
      function Ae(y, z, ce) {
        var V = "";
        typeof y == "string" && (y = JSON.parse(y)), typeof z == "string" && (z = JSON.parse(z));
        var se = Array.isArray(y) && 0 < y.length, ie = !Array.isArray(z[0]);
        if (se && _) {
          for (var h = 0; h < y.length; h++)
            0 < h && (V += m), V += fe(y[h], h);
          0 < z.length && (V += S);
        }
        for (var b = 0; b < z.length; b++) {
          var N = se ? y.length : z[b].length, D = !1, re = se ? Object.keys(z[b]).length === 0 : z[b].length === 0;
          if (ce && !se && (D = ce === "greedy" ? z[b].join("").trim() === "" : z[b].length === 1 && z[b][0].length === 0), ce === "greedy" && se) {
            for (var H = [], le = 0; le < N; le++) {
              var X = ie ? y[le] : le;
              H.push(z[b][X]);
            }
            D = H.join("").trim() === "";
          }
          if (!D) {
            for (var G = 0; G < N; G++) {
              0 < G && !re && (V += m);
              var me = se && ie ? y[G] : G;
              V += fe(z[b][me], G);
            }
            b < z.length - 1 && (!ce || 0 < N && !re) && (V += S);
          }
        }
        return V;
      }
      function fe(y, z) {
        if (y == null)
          return "";
        if (y.constructor === Date)
          return JSON.stringify(y).slice(1, 25);
        var ce = !1;
        ne && typeof y == "string" && ne.test(y) && (y = "'" + y, ce = !0);
        var V = y.toString().replace(x, ae);
        return (ce = ce || f === !0 || typeof f == "function" && f(y, z) || Array.isArray(f) && f[z] || function(se, ie) {
          for (var h = 0; h < ie.length; h++)
            if (-1 < se.indexOf(ie[h]))
              return !0;
          return !1;
        }(V, E.BAD_DELIMITERS) || -1 < V.indexOf(m) || V.charAt(0) === " " || V.charAt(V.length - 1) === " ") ? F + V + F : V;
      }
    } };
    if (E.RECORD_SEP = "", E.UNIT_SEP = "", E.BYTE_ORDER_MARK = "\uFEFF", E.BAD_DELIMITERS = ["\r", `
`, '"', E.BYTE_ORDER_MARK], E.WORKERS_SUPPORTED = !U && !!g.Worker, E.NODE_STREAM_INPUT = 1, E.LocalChunkSize = 10485760, E.RemoteChunkSize = 5242880, E.DefaultDelimiter = ",", E.Parser = Ue, E.ParserHandle = et, E.NetworkStreamer = R, E.FileStreamer = P, E.StringStreamer = j, E.ReadableStreamStreamer = Fe, g.jQuery) {
      var C = g.jQuery;
      C.fn.parse = function(a) {
        var i = a.config || {}, f = [];
        return this.each(function(S) {
          if (!(C(this).prop("tagName").toUpperCase() === "INPUT" && C(this).attr("type").toLowerCase() === "file" && g.FileReader) || !this.files || this.files.length === 0)
            return !0;
          for (var F = 0; F < this.files.length; F++)
            f.push({ file: this.files[F], inputElem: this, instanceConfig: C.extend({}, i) });
        }), _(), this;
        function _() {
          if (f.length !== 0) {
            var S, F, ae, Z, w = f[0];
            if (B(a.before)) {
              var ne = a.before(w.file, w.inputElem);
              if (typeof ne == "object") {
                if (ne.action === "abort")
                  return S = "AbortError", F = w.file, ae = w.inputElem, Z = ne.reason, void (B(a.error) && a.error({ name: S }, F, ae, Z));
                if (ne.action === "skip")
                  return void m();
                typeof ne.config == "object" && (w.instanceConfig = C.extend(w.instanceConfig, ne.config));
              } else if (ne === "skip")
                return void m();
            }
            var x = w.instanceConfig.complete;
            w.instanceConfig.complete = function(Ae) {
              B(x) && x(Ae, w.file, w.inputElem), m();
            }, E.parse(w.file, w.instanceConfig);
          } else
            B(a.complete) && a.complete();
        }
        function m() {
          f.splice(0, 1), _();
        }
      };
    }
    function d(a) {
      this._handle = null, this._finished = !1, this._completed = !1, this._halted = !1, this._input = null, this._baseIndex = 0, this._partialLine = "", this._rowCount = 0, this._start = 0, this._nextChunk = null, this.isFirstChunk = !0, this._completeResults = { data: [], errors: [], meta: {} }, (function(i) {
        var f = Pe(i);
        f.chunkSize = parseInt(f.chunkSize), i.step || i.chunk || (f.chunkSize = null), this._handle = new et(f), (this._handle.streamer = this)._config = f;
      }).call(this, a), this.parseChunk = function(i, f) {
        if (this.isFirstChunk && B(this._config.beforeFirstChunk)) {
          var _ = this._config.beforeFirstChunk(i);
          _ !== void 0 && (i = _);
        }
        this.isFirstChunk = !1, this._halted = !1;
        var m = this._partialLine + i;
        this._partialLine = "";
        var S = this._handle.parse(m, this._baseIndex, !this._finished);
        if (!this._handle.paused() && !this._handle.aborted()) {
          var F = S.meta.cursor;
          this._finished || (this._partialLine = m.substring(F - this._baseIndex), this._baseIndex = F), S && S.data && (this._rowCount += S.data.length);
          var ae = this._finished || this._config.preview && this._rowCount >= this._config.preview;
          if (W)
            g.postMessage({ results: S, workerId: E.WORKER_ID, finished: ae });
          else if (B(this._config.chunk) && !f) {
            if (this._config.chunk(S, this._handle), this._handle.paused() || this._handle.aborted())
              return void (this._halted = !0);
            S = void 0, this._completeResults = void 0;
          }
          return this._config.step || this._config.chunk || (this._completeResults.data = this._completeResults.data.concat(S.data), this._completeResults.errors = this._completeResults.errors.concat(S.errors), this._completeResults.meta = S.meta), this._completed || !ae || !B(this._config.complete) || S && S.meta.aborted || (this._config.complete(this._completeResults, this._input), this._completed = !0), ae || S && S.meta.paused || this._nextChunk(), S;
        }
        this._halted = !0;
      }, this._sendError = function(i) {
        B(this._config.error) ? this._config.error(i) : W && this._config.error && g.postMessage({ workerId: E.WORKER_ID, error: i, finished: !1 });
      };
    }
    function R(a) {
      var i;
      (a = a || {}).chunkSize || (a.chunkSize = E.RemoteChunkSize), d.call(this, a), this._nextChunk = U ? function() {
        this._readChunk(), this._chunkLoaded();
      } : function() {
        this._readChunk();
      }, this.stream = function(f) {
        this._input = f, this._nextChunk();
      }, this._readChunk = function() {
        if (this._finished)
          this._chunkLoaded();
        else {
          if (i = new XMLHttpRequest(), this._config.withCredentials && (i.withCredentials = this._config.withCredentials), U || (i.onload = Se(this._chunkLoaded, this), i.onerror = Se(this._chunkError, this)), i.open(this._config.downloadRequestBody ? "POST" : "GET", this._input, !U), this._config.downloadRequestHeaders) {
            var f = this._config.downloadRequestHeaders;
            for (var _ in f)
              i.setRequestHeader(_, f[_]);
          }
          if (this._config.chunkSize) {
            var m = this._start + this._config.chunkSize - 1;
            i.setRequestHeader("Range", "bytes=" + this._start + "-" + m);
          }
          try {
            i.send(this._config.downloadRequestBody);
          } catch (S) {
            this._chunkError(S.message);
          }
          U && i.status === 0 && this._chunkError();
        }
      }, this._chunkLoaded = function() {
        i.readyState === 4 && (i.status < 200 || 400 <= i.status ? this._chunkError() : (this._start += this._config.chunkSize ? this._config.chunkSize : i.responseText.length, this._finished = !this._config.chunkSize || this._start >= function(f) {
          var _ = f.getResponseHeader("Content-Range");
          return _ === null ? -1 : parseInt(_.substring(_.lastIndexOf("/") + 1));
        }(i), this.parseChunk(i.responseText)));
      }, this._chunkError = function(f) {
        var _ = i.statusText || f;
        this._sendError(new Error(_));
      };
    }
    function P(a) {
      var i, f;
      (a = a || {}).chunkSize || (a.chunkSize = E.LocalChunkSize), d.call(this, a);
      var _ = typeof FileReader < "u";
      this.stream = function(m) {
        this._input = m, f = m.slice || m.webkitSlice || m.mozSlice, _ ? ((i = new FileReader()).onload = Se(this._chunkLoaded, this), i.onerror = Se(this._chunkError, this)) : i = new FileReaderSync(), this._nextChunk();
      }, this._nextChunk = function() {
        this._finished || this._config.preview && !(this._rowCount < this._config.preview) || this._readChunk();
      }, this._readChunk = function() {
        var m = this._input;
        if (this._config.chunkSize) {
          var S = Math.min(this._start + this._config.chunkSize, this._input.size);
          m = f.call(m, this._start, S);
        }
        var F = i.readAsText(m, this._config.encoding);
        _ || this._chunkLoaded({ target: { result: F } });
      }, this._chunkLoaded = function(m) {
        this._start += this._config.chunkSize, this._finished = !this._config.chunkSize || this._start >= this._input.size, this.parseChunk(m.target.result);
      }, this._chunkError = function() {
        this._sendError(i.error);
      };
    }
    function j(a) {
      var i;
      d.call(this, a = a || {}), this.stream = function(f) {
        return i = f, this._nextChunk();
      }, this._nextChunk = function() {
        if (!this._finished) {
          var f, _ = this._config.chunkSize;
          return _ ? (f = i.substring(0, _), i = i.substring(_)) : (f = i, i = ""), this._finished = !i, this.parseChunk(f);
        }
      };
    }
    function Fe(a) {
      d.call(this, a = a || {});
      var i = [], f = !0, _ = !1;
      this.pause = function() {
        d.prototype.pause.apply(this, arguments), this._input.pause();
      }, this.resume = function() {
        d.prototype.resume.apply(this, arguments), this._input.resume();
      }, this.stream = function(m) {
        this._input = m, this._input.on("data", this._streamData), this._input.on("end", this._streamEnd), this._input.on("error", this._streamError);
      }, this._checkIsFinished = function() {
        _ && i.length === 1 && (this._finished = !0);
      }, this._nextChunk = function() {
        this._checkIsFinished(), i.length ? this.parseChunk(i.shift()) : f = !0;
      }, this._streamData = Se(function(m) {
        try {
          i.push(typeof m == "string" ? m : m.toString(this._config.encoding)), f && (f = !1, this._checkIsFinished(), this.parseChunk(i.shift()));
        } catch (S) {
          this._streamError(S);
        }
      }, this), this._streamError = Se(function(m) {
        this._streamCleanUp(), this._sendError(m);
      }, this), this._streamEnd = Se(function() {
        this._streamCleanUp(), _ = !0, this._streamData("");
      }, this), this._streamCleanUp = Se(function() {
        this._input.removeListener("data", this._streamData), this._input.removeListener("end", this._streamEnd), this._input.removeListener("error", this._streamError);
      }, this);
    }
    function et(a) {
      var i, f, _, m = Math.pow(2, 53), S = -m, F = /^\s*-?(\d+\.?|\.\d+|\d+\.\d+)([eE][-+]?\d+)?\s*$/, ae = /^((\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z))|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z))|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d([+-][0-2]\d:[0-5]\d|Z)))$/, Z = this, w = 0, ne = 0, x = !1, Ae = !1, fe = [], y = { data: [], errors: [], meta: {} };
      if (B(a.step)) {
        var z = a.step;
        a.step = function(b) {
          if (y = b, se())
            V();
          else {
            if (V(), y.data.length === 0)
              return;
            w += b.data.length, a.preview && w > a.preview ? f.abort() : (y.data = y.data[0], z(y, Z));
          }
        };
      }
      function ce(b) {
        return a.skipEmptyLines === "greedy" ? b.join("").trim() === "" : b.length === 1 && b[0].length === 0;
      }
      function V() {
        return y && _ && (h("Delimiter", "UndetectableDelimiter", "Unable to auto-detect delimiting character; defaulted to '" + E.DefaultDelimiter + "'"), _ = !1), a.skipEmptyLines && (y.data = y.data.filter(function(b) {
          return !ce(b);
        })), se() && function() {
          if (!y)
            return;
          function b(D, re) {
            B(a.transformHeader) && (D = a.transformHeader(D, re)), fe.push(D);
          }
          if (Array.isArray(y.data[0])) {
            for (var N = 0; se() && N < y.data.length; N++)
              y.data[N].forEach(b);
            y.data.splice(0, 1);
          } else
            y.data.forEach(b);
        }(), function() {
          if (!y || !a.header && !a.dynamicTyping && !a.transform)
            return y;
          function b(D, re) {
            var H, le = a.header ? {} : [];
            for (H = 0; H < D.length; H++) {
              var X = H, G = D[H];
              a.header && (X = H >= fe.length ? "__parsed_extra" : fe[H]), a.transform && (G = a.transform(G, X)), G = ie(X, G), X === "__parsed_extra" ? (le[X] = le[X] || [], le[X].push(G)) : le[X] = G;
            }
            return a.header && (H > fe.length ? h("FieldMismatch", "TooManyFields", "Too many fields: expected " + fe.length + " fields but parsed " + H, ne + re) : H < fe.length && h("FieldMismatch", "TooFewFields", "Too few fields: expected " + fe.length + " fields but parsed " + H, ne + re)), le;
          }
          var N = 1;
          return !y.data.length || Array.isArray(y.data[0]) ? (y.data = y.data.map(b), N = y.data.length) : y.data = b(y.data, 0), a.header && y.meta && (y.meta.fields = fe), ne += N, y;
        }();
      }
      function se() {
        return a.header && fe.length === 0;
      }
      function ie(b, N) {
        return D = b, a.dynamicTypingFunction && a.dynamicTyping[D] === void 0 && (a.dynamicTyping[D] = a.dynamicTypingFunction(D)), (a.dynamicTyping[D] || a.dynamicTyping) === !0 ? N === "true" || N === "TRUE" || N !== "false" && N !== "FALSE" && (function(re) {
          if (F.test(re)) {
            var H = parseFloat(re);
            if (S < H && H < m)
              return !0;
          }
          return !1;
        }(N) ? parseFloat(N) : ae.test(N) ? new Date(N) : N === "" ? null : N) : N;
        var D;
      }
      function h(b, N, D, re) {
        var H = { type: b, code: N, message: D };
        re !== void 0 && (H.row = re), y.errors.push(H);
      }
      this.parse = function(b, N, D) {
        var re = a.quoteChar || '"';
        if (a.newline || (a.newline = function(X, G) {
          X = X.substring(0, 1048576);
          var me = new RegExp(he(G) + "([^]*?)" + he(G), "gm"), we = (X = X.replace(me, "")).split("\r"), Ie = X.split(`
`), Le = 1 < Ie.length && Ie[0].length < we[0].length;
          if (we.length === 1 || Le)
            return `
`;
          for (var Ee = 0, Y = 0; Y < we.length; Y++)
            we[Y][0] === `
` && Ee++;
          return Ee >= we.length / 2 ? `\r
` : "\r";
        }(b, re)), _ = !1, a.delimiter)
          B(a.delimiter) && (a.delimiter = a.delimiter(b), y.meta.delimiter = a.delimiter);
        else {
          var H = function(X, G, me, we, Ie) {
            var Le, Ee, Y, ee;
            Ie = Ie || [",", "	", "|", ";", E.RECORD_SEP, E.UNIT_SEP];
            for (var De = 0; De < Ie.length; De++) {
              var $ = Ie[De], ke = 0, Me = 0, Ge = 0;
              Y = void 0;
              for (var Oe = new Ue({ comments: we, delimiter: $, newline: G, preview: 10 }).parse(X), be = 0; be < Oe.data.length; be++)
                if (me && ce(Oe.data[be]))
                  Ge++;
                else {
                  var xe = Oe.data[be].length;
                  Me += xe, Y !== void 0 ? 0 < xe && (ke += Math.abs(xe - Y), Y = xe) : Y = xe;
                }
              0 < Oe.data.length && (Me /= Oe.data.length - Ge), (Ee === void 0 || ke <= Ee) && (ee === void 0 || ee < Me) && 1.99 < Me && (Ee = ke, Le = $, ee = Me);
            }
            return { successful: !!(a.delimiter = Le), bestDelimiter: Le };
          }(b, a.newline, a.skipEmptyLines, a.comments, a.delimitersToGuess);
          H.successful ? a.delimiter = H.bestDelimiter : (_ = !0, a.delimiter = E.DefaultDelimiter), y.meta.delimiter = a.delimiter;
        }
        var le = Pe(a);
        return a.preview && a.header && le.preview++, i = b, f = new Ue(le), y = f.parse(i, N, D), V(), x ? { meta: { paused: !0 } } : y || { meta: { paused: !1 } };
      }, this.paused = function() {
        return x;
      }, this.pause = function() {
        x = !0, f.abort(), i = B(a.chunk) ? "" : i.substring(f.getCharIndex());
      }, this.resume = function() {
        Z.streamer._halted ? (x = !1, Z.streamer.parseChunk(i, !0)) : setTimeout(Z.resume, 3);
      }, this.aborted = function() {
        return Ae;
      }, this.abort = function() {
        Ae = !0, f.abort(), y.meta.aborted = !0, B(a.complete) && a.complete(y), i = "";
      };
    }
    function he(a) {
      return a.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
    }
    function Ue(a) {
      var i, f = (a = a || {}).delimiter, _ = a.newline, m = a.comments, S = a.step, F = a.preview, ae = a.fastMode, Z = i = a.quoteChar === void 0 || a.quoteChar === null ? '"' : a.quoteChar;
      if (a.escapeChar !== void 0 && (Z = a.escapeChar), (typeof f != "string" || -1 < E.BAD_DELIMITERS.indexOf(f)) && (f = ","), m === f)
        throw new Error("Comment character same as delimiter");
      m === !0 ? m = "#" : (typeof m != "string" || -1 < E.BAD_DELIMITERS.indexOf(m)) && (m = !1), _ !== `
` && _ !== "\r" && _ !== `\r
` && (_ = `
`);
      var w = 0, ne = !1;
      this.parse = function(x, Ae, fe) {
        if (typeof x != "string")
          throw new Error("Input must be a string");
        var y = x.length, z = f.length, ce = _.length, V = m.length, se = B(S), ie = [], h = [], b = [], N = w = 0;
        if (!x)
          return _e();
        if (a.header && !Ae) {
          var D = x.split(_)[0].split(f), re = [], H = {}, le = !1;
          for (var X in D) {
            var G = D[X];
            B(a.transformHeader) && (G = a.transformHeader(G, X));
            var me = G, we = H[G] || 0;
            for (0 < we && (le = !0, me = G + "_" + we), H[G] = we + 1; re.includes(me); )
              me = me + "_" + we;
            re.push(me);
          }
          if (le) {
            var Ie = x.split(_);
            Ie[0] = re.join(f), x = Ie.join(_);
          }
        }
        if (ae || ae !== !1 && x.indexOf(i) === -1) {
          for (var Le = x.split(_), Ee = 0; Ee < Le.length; Ee++) {
            if (b = Le[Ee], w += b.length, Ee !== Le.length - 1)
              w += _.length;
            else if (fe)
              return _e();
            if (!m || b.substring(0, V) !== m) {
              if (se) {
                if (ie = [], Ge(b.split(f)), Te(), ne)
                  return _e();
              } else
                Ge(b.split(f));
              if (F && F <= Ee)
                return ie = ie.slice(0, F), _e(!0);
            }
          }
          return _e();
        }
        for (var Y = x.indexOf(f, w), ee = x.indexOf(_, w), De = new RegExp(he(Z) + he(i), "g"), $ = x.indexOf(i, w); ; )
          if (x[w] !== i)
            if (m && b.length === 0 && x.substring(w, w + V) === m) {
              if (ee === -1)
                return _e();
              w = ee + ce, ee = x.indexOf(_, w), Y = x.indexOf(f, w);
            } else if (Y !== -1 && (Y < ee || ee === -1))
              b.push(x.substring(w, Y)), w = Y + z, Y = x.indexOf(f, w);
            else {
              if (ee === -1)
                break;
              if (b.push(x.substring(w, ee)), xe(ee + ce), se && (Te(), ne))
                return _e();
              if (F && ie.length >= F)
                return _e(!0);
            }
          else
            for ($ = w, w++; ; ) {
              if (($ = x.indexOf(i, $ + 1)) === -1)
                return fe || h.push({ type: "Quotes", code: "MissingQuotes", message: "Quoted field unterminated", row: ie.length, index: w }), be();
              if ($ === y - 1)
                return be(x.substring(w, $).replace(De, i));
              if (i !== Z || x[$ + 1] !== Z) {
                if (i === Z || $ === 0 || x[$ - 1] !== Z) {
                  Y !== -1 && Y < $ + 1 && (Y = x.indexOf(f, $ + 1)), ee !== -1 && ee < $ + 1 && (ee = x.indexOf(_, $ + 1));
                  var ke = Oe(ee === -1 ? Y : Math.min(Y, ee));
                  if (x.substr($ + 1 + ke, z) === f) {
                    b.push(x.substring(w, $).replace(De, i)), x[w = $ + 1 + ke + z] !== i && ($ = x.indexOf(i, w)), Y = x.indexOf(f, w), ee = x.indexOf(_, w);
                    break;
                  }
                  var Me = Oe(ee);
                  if (x.substring($ + 1 + Me, $ + 1 + Me + ce) === _) {
                    if (b.push(x.substring(w, $).replace(De, i)), xe($ + 1 + Me + ce), Y = x.indexOf(f, w), $ = x.indexOf(i, w), se && (Te(), ne))
                      return _e();
                    if (F && ie.length >= F)
                      return _e(!0);
                    break;
                  }
                  h.push({ type: "Quotes", code: "InvalidQuotes", message: "Trailing quote on quoted field is malformed", row: ie.length, index: w }), $++;
                }
              } else
                $++;
            }
        return be();
        function Ge(q) {
          ie.push(q), N = w;
        }
        function Oe(q) {
          var $e = 0;
          if (q !== -1) {
            var ft = x.substring($ + 1, q);
            ft && ft.trim() === "" && ($e = ft.length);
          }
          return $e;
        }
        function be(q) {
          return fe || (q === void 0 && (q = x.substring(w)), b.push(q), w = y, Ge(b), se && Te()), _e();
        }
        function xe(q) {
          w = q, Ge(b), b = [], ee = x.indexOf(_, w);
        }
        function _e(q) {
          return { data: ie, errors: h, meta: { delimiter: f, linebreak: _, aborted: ne, truncated: !!q, cursor: N + (Ae || 0) } };
        }
        function Te() {
          S(_e()), ie = [], h = [];
        }
      }, this.abort = function() {
        ne = !0;
      }, this.getCharIndex = function() {
        return w;
      };
    }
    function pe(a) {
      var i = a.data, f = L[i.workerId], _ = !1;
      if (i.error)
        f.userError(i.error, i.file);
      else if (i.results && i.results.data) {
        var m = { abort: function() {
          _ = !0, Ce(i.workerId, { data: [], errors: [], meta: { aborted: !0 } });
        }, pause: ut, resume: ut };
        if (B(f.userStep)) {
          for (var S = 0; S < i.results.data.length && (f.userStep({ data: i.results.data[S], errors: i.results.errors, meta: i.results.meta }, m), !_); S++)
            ;
          delete i.results;
        } else
          B(f.userChunk) && (f.userChunk(i.results, m, i.file), delete i.results);
      }
      i.finished && !_ && Ce(i.workerId, i.results);
    }
    function Ce(a, i) {
      var f = L[a];
      B(f.userComplete) && f.userComplete(i), f.terminate(), delete L[a];
    }
    function ut() {
      throw new Error("Not implemented.");
    }
    function Pe(a) {
      if (typeof a != "object" || a === null)
        return a;
      var i = Array.isArray(a) ? [] : {};
      for (var f in a)
        i[f] = Pe(a[f]);
      return i;
    }
    function Se(a, i) {
      return function() {
        a.apply(i, arguments);
      };
    }
    function B(a) {
      return typeof a == "function";
    }
    return W && (g.onmessage = function(a) {
      var i = a.data;
      if (E.WORKER_ID === void 0 && i && (E.WORKER_ID = i.workerId), typeof i.input == "string")
        g.postMessage({ workerId: E.WORKER_ID, results: E.parse(i.input, i.config), finished: !0 });
      else if (g.File && i.input instanceof File || i.input instanceof Object) {
        var f = E.parse(i.input, i.config);
        f && g.postMessage({ workerId: E.WORKER_ID, results: f, finished: !0 });
      }
    }), (R.prototype = Object.create(d.prototype)).constructor = R, (P.prototype = Object.create(d.prototype)).constructor = P, (j.prototype = Object.create(j.prototype)).constructor = j, (Fe.prototype = Object.create(d.prototype)).constructor = Fe, E;
  });
})(rr);
var Wr = rr.exports;
const jr = ["id"], Jr = /* @__PURE__ */ ir({
  __name: "JsonToCSV",
  props: {
    data: { default: [] },
    fields: { default: void 0 },
    name: { default: "export.csv" },
    delimiter: { default: "," },
    separatorExcel: { type: Boolean, default: !1 },
    encoding: { default: "utf-8" },
    advancedOptions: { default: {} },
    labels: { default: void 0 },
    testing: { type: Boolean, default: !1 },
    id: { default: "button" },
    buttonName: { default: "button" }
  },
  emits: ["export-started", "export-finished"],
  setup(c, { emit: O }) {
    const A = (d, R) => typeof d === R, g = c, U = O;
    "" + (/* @__PURE__ */ new Date()).getTime();
    const W = () => {
      let d = g.labels;
      if (!A(d, "undefined") && !A(d, "function") && !A(d, "object"))
        throw new Error("Labels needs to be a function(value, key) or object.");
      return A(d, "function") ? (R) => Xn(R, d) : A(d, "object") ? (R) => Xn(R, (P, j) => d[j] || j) : (R) => R;
    }, L = () => {
      let d = g.fields;
      if (!A(d, "undefined") && !A(d, "function") && !A(d, "object") && !Array.isArray(d))
        throw new Error("Fields needs to be a function(value, key) or array.");
      return A(d, "function") || A(d, "object") && !Array.isArray(d) ? (R) => hr(R, d) : Array.isArray(d) ? (R) => $r(R, d) : (R) => R;
    }, ue = () => {
      if (A(g.fields, "undefined") && A(g.labels, "undefined"))
        return g.data;
      const d = W(), R = L();
      return g.data.map((P) => d(R(P)));
    }, E = ar(() => {
      const d = ue();
      return d.length ? d : null;
    }), C = () => {
      U("export-started");
      const d = E.value;
      if (!d) {
        console.error("No data to export");
        return;
      }
      let R = Wr.unparse(
        d,
        Object.assign(
          {
            delimiter: g.delimiter,
            encoding: g.encoding
          },
          g.advancedOptions
        )
      );
      if (g.separatorExcel && (R = "SEP=" + g.delimiter + `\r
` + R), g.encoding === "utf-8" && (R = "\uFEFF" + R), U("export-finished"), !g.testing) {
        let P = new Blob([R], {
          type: "text/csv;charset=" + g.encoding
        });
        qr.saveAs(P, g.name);
      }
    };
    return (d, R) => (sr(), or("div", {
      id: d.id,
      onClick: C
    }, [
      ur(d.$slots, "default", {}, () => [
        fr(cr(d.buttonName), 1)
      ])
    ], 8, jr));
  }
});
export {
  Jr as default
};
